pallets/gear-eth-bridge/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+// This file is part of Gear.
+
+// Copyright (C) 2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Pallet Gear Eth Bridge.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![doc(html_favicon_url = "https://gear-tech.io/favicons/favicon.ico")]
+#![doc(html_logo_url = "https://docs.gear.rs/logo.svg")]
+#![warn(missing_docs)]
+
+pub use builtin::Actor;
+pub use internal::{EthMessage, Proof};
+pub use pallet::*;
+
+mod builtin;
+mod internal;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[allow(missing_docs)]
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use common::Origin;
+    use frame_support::{
+        pallet_prelude::*,
+        traits::{ConstBool, OneSessionHandler, StorageVersion},
+        StorageHasher,
+    };
+    use frame_system::{
+        ensure_root, ensure_signed,
+        pallet_prelude::{BlockNumberFor, OriginFor},
+    };
+    use gprimitives::{ActorId, H160, H256, U256};
+    use sp_runtime::{
+        traits::{Keccak256, One, Saturating, Zero},
+        BoundToRuntimeAppPublic, RuntimeAppPublic,
+    };
+    use sp_std::vec::Vec;
+
+    type QueueCapacityOf<T> = <T as Config>::QueueCapacity;
+    type SessionsPerEraOf<T> = <T as Config>::SessionsPerEra;
+
+    /// Pallet Gear Eth Bridge's storage version.
+    pub const ETH_BRIDGE_STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+    /// Pallet Gear Eth Bridge's config.
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_staking::Config {
```

Comment:
Please help me see why the `Config` trait should also extend the `pallet_staking::Config`